### PR TITLE
Api further develop

### PR DESCRIPTION
**Pull request by henryletton**
Filling some API data gaps.
Also a few other small tweaks.

### CODE REVIEW

File: server/src/main/java/edu/uob/PostOperations.java [hunk 1]
```diff
@@ -89,4 +89,43 @@ public static ResponseEntity<ObjectNode> postAccount(String username, String ema
...
+            try(PreparedStatement s = c.prepareStatement(SQL)) {
+                s.setString(1, startDate);
+                s.setString(2, endDate);
+                ResultSet r = s.executeQuery();
+                r.next();
+                datesExist = r.getBoolean(1);
+            }
+            // Yes, set only to true
+            if(datesExist) {
+                SQL = "UPDATE rotagroups SET status = false WHERE 1=1; " +
```
Comment (epines1911) [inline server/src/main/java/edu/uob/PostOperations.java:108]: I am just curious, why do we need to write "WHERE 1=1" here? 🤔

File: server/src/test/java/edu/uob/PutOperationsTests.java [hunk 2]
```diff
@@ -143,4 +148,58 @@ public void testPutFixedShift() {
...
+    @Test
+    public void testPutAccountRotaType() throws JsonProcessingException {
+        // Get random id and username to test
+        int id1 = TestTools.getTestAccountId();
+        String username1 = TestTools.getRandomUsername();
+        String connectionString = ConnectionTools.getConnectionString();
+        try(Connection c = DriverManager.getConnection(connectionString)) {
+            // Create new account with id (definitely unused)
+            assertFalse(ConnectionTools.accountIdExists(id1, c));
+            String SQL = "INSERT INTO accounts (id, username, password, email) " +
```
Comment (epines1911) [inline server/src/test/java/edu/uob/PutOperationsTests.java:161]: @epines1911 todo use a random email address to replace the email here

File: server/src/main/java/edu/uob/PostOperations.java (hunk 1 above)
Comment (henryletton) [reply server/src/main/java/edu/uob/PostOperations.java:108]: Running an update query without a where clause can sometimes cause a warning or error. 
"Unsafe query: 'Update' statement without 'where' updates all table rows at once" - warning from DataGrip.
Including "WHERE 1=1" removes this potential issue, although achieves the same thing as it is true for all rows.

Comment (epines1911) [reply server/src/main/java/edu/uob/PostOperations.java:108]: Thanks for your explanation!

**Pull request merged**